File: redditScraper.py
import praw
import datetime
import json

client_id_str = 'vJpj1KFoozVc_A'
client_secret_str = 'YR9SRuIqLVxBm7uNaZiGM60Jylw'

reddit = praw.Reddit(user_agent='Comment Extraction (by /u/mosma)',
                     client_id=client_id_str, client_secret=client_secret_str)
def extract_subreddit(subreddit_str = 'politics', output_filename = 'out.json'):
    subreddit = praw.models.Subreddit(reddit,subreddit_str)
    submissions = subreddit.top(time_filter = 'day')
    submissions_list = []
    for submission in submissions:
        submission.comments.replace_more()
        url = "http://www.reddit.com" + submission.permalink
        title = submission.title
        author = str(submission.author)
        time = str(datetime.datetime.fromtimestamp(submission.created))
        link = submission.url
        num_comments = submission.num_comments
        comments = [{'author':str(comment.author),'id':comment.id,
                     'parent_id':comment.parent_id,'body':str(comment.body),
                     'score':comment.score} for comment in submission.comments.list()]
        submission_dict = {'url':url,'title':title,'author':author,'time':time,'link':link,'num_comments':num_comments,'comments':comments}
        submissions_list.append(submission_dict)
    json_obj = json.dumps(submissions_list,indent = 4,separators=(',', ': '))
    with open(subreddit_str + '_' + output_filename,'w') as outfile:
        json.dump(submissions_list,outfile,indent = 4,separators=(',', ': '))
    return json_obj

extract_subreddit('books')
    

File: redditScraper.js
/**
 * Created by Moshe on 25/02/2017.
 */
'use strict'
/*var uniq = require('uniq');
var nums = [ 5, 2, 1, 3, 2, 5, 4, 2, 0, 1 ];
console.log(uniq(nums));*/

const snoowrap = require('snoowrap');

var user_agent_val = 'Comment Extraction by /u/mosma';
var client_id_val = 'vJpj1KFoozVc_A';
var client_secret_val = 'YR9SRuIqLVxBm7uNaZiGM60Jylw';
var username_val = 'mosma';
var password_val = 'themumin09';

const config = {
    user_agent: user_agent_val,
    client_id: client_id_val,
    client_secret: client_secret_val,
    username: username_val,
    password: password_val
}

const r = new snoowrap(config);

var submission_id = '5rrasx';
r.getSubmission(submission_id).expandReplies({limit:5, depth:5}).then(console.log);
